source/framefmt.c

```diff
@@ -235,16 +254,26 @@ int aws_cryptosdk_serialize_frame(
     size_t plaintext_size,
     struct aws_byte_buf *ciphertext_buf,
     const struct aws_cryptosdk_alg_properties *alg_props) {
+    AWS_PRECONDITION(aws_cryptosdk_alg_properties_is_valid(alg_props));
     struct aws_cryptosdk_framestate state;
 
+    // The plaintext_size should be bound to prevent arithmetic
+    // overflows due to addition
+    if ((frame->type == FRAME_TYPE_SINGLE && plaintext_size > MAX_UNFRAMED_PLAINTEXT_SIZE) ||
```

Comment:
Are these bounds documented in the documentation @ttjsu-aws

Reply:
Yes, we currently have that: 

- The total bytes allowed in a single frame MUST be less than or equal to 2^32 - 1.
- The number of frames in a single message MUST be less than or equal to 2^32 - 1.
- The encrypted content length MUST NOT be greater than 2^36 - 32, or 64 gibibytes (64 GiB).